scripts/update-master-formula.sh

```diff
@@ -1,19 +1,33 @@
 #!/bin/bash
-# updates a formula on the master server
+# updates a formula on a masterless or master server
 
 set -e # everything must pass
 set -u # no unbound variables
 
 pname=$1
 formula=$2
+revision=${3:-""}
 
-mkdir -p /opt/formulas
-if [ -d "/opt/formulas/$pname" ]; then
-    cd "/opt/formulas/$pname"
+formula_root="/opt/formulas"
+if [ "$pname" == "builder-private" ]; then
+    formula_root="/opt"
+fi
+formula_path="$formula_root/$pname"
+
+mkdir -p "$formula_root"
+
+if [ -d "$formula_path" ]; then
+    cd "$formula_path"
     git reset --hard
     git clean -d --force
     git pull --rebase
 else
-    cd /opt/formulas/
+    cd "$formula_root"
     git clone "$formula" "$pname"
 fi
+
+cd "$formula_path"
+
+if [ "$revision" != "" ]; then
```

Comment:
may need a `git fetch` before, it this performed previously?
Should be needed in all cases where the commit to be checked out is not in the current branch